cucumber-core/src/main/java/io/cucumber/core/runner/TestAbortedExceptions.java

```diff
@@ -26,20 +26,18 @@ final class TestAbortedExceptions {
 
     static Predicate<Throwable> createIsTestAbortedExceptionPredicate() {
         ClassLoader defaultClassLoader = ClassLoaders.getDefaultClassLoader();
-        return Arrays.stream(TEST_ABORTED_EXCEPTIONS)
-                .flatMap(s -> {
+        return throwable -> Arrays.stream(TEST_ABORTED_EXCEPTIONS)
```

Comment:
Would a for loop be faster?

Reply:
I benchmarked the following code:

static Predicate<Throwable> createIsTestAbortedExceptionPredicate7() {
        ClassLoader defaultClassLoader = ClassLoaders.getDefaultClassLoader();
        return throwable -> {
            for (String s : TEST_ABORTED_EXCEPTIONS) {
                try {
                    Class<?> aClass = defaultClassLoader.loadClass(s);
                    if (aClass.isInstance(throwable)) {
                        return true;
                    }
                } catch (Throwable t) {
                    rethrowIfUnrecoverable(t);
                    log.debug(t,
                            () -> String.format(
                                    "Failed to load class %s: will not be supported for aborted executions.", s));
                }
            }
            return false;
        };
    }

The JMH benchmark says the variant 7 (the one with a loop) is slightly slower than the proposed version (variant 6), but not significantly :

Benchmark                                                               Mode  Cnt          Score          Error  Units
TestAbortedExceptionsBenchmark.createIsTestAbortedExceptionPredicate5  thrpt   25  264824252,996 ±  4775342,657  ops/s
TestAbortedExceptionsBenchmark.createIsTestAbortedExceptionPredicate6  thrpt   25  259886661,320 ±  1766902,768  ops/s
TestAbortedExceptionsBenchmark.createIsTestAbortedExceptionPredicate7  thrpt   25  254627500,665 ± 12269987,245  ops/s

I think the code readability is better with the Stream+anyMatch than with the loop, so I would prefer to keep the proposed version..